File: backend/app/schemas/stock.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel


class StockResponse(BaseModel):
    code: str
    name: str
    exchange: Optional[str] = None
    industry: Optional[str] = None

    class Config:
        from_attributes = True


class QuoteResponse(BaseModel):
    code: str
    name: str
    price: float
    change: float
    change_pct: float
    open: float
    high: float
    low: float
    volume: float
    amount: float


class DailyBarResponse(BaseModel):
    date: date
    open: float
    high: float
    low: float
    close: float
    volume: Optional[float] = None
    amount: Optional[float] = None


class IndexResponse(BaseModel):
    code: str
    name: str
    price: float
    change: float
    change_pct: float


class WatchlistCreate(BaseModel):
    code: str
    name: str
    group_name: str = "默认"


class WatchlistUpdate(BaseModel):
    sort_order: Optional[int] = None
    group_name: Optional[str] = None


class WatchlistResponse(BaseModel):
    id: int
    code: str
    name: str
    added_at: datetime
    sort_order: int
    group_name: str

    class Config:
        from_attributes = True

File: frontend/src/types/stock.ts
export interface IndexData {
  code: string
  name: string
  price: number
  change: number
  change_pct: number
}

export interface QuoteData {
  code: string
  name: string
  price: number
  change: number
  change_pct: number
  open: number
  high: number
  low: number
  volume: number
  amount: number
}

export interface BarData {
  date: string
  open: number
  high: number
  low: number
  close: number
  volume: number | null
  amount: number | null
}

export interface StockInfo {
  code: string
  name: string
  exchange?: string
  industry?: string
}

export interface WatchlistItem {
  id: number
  code: string
  name: string
  added_at: string
  sort_order: number
  group_name: string
}

export interface AlertRule {
  id: number
  stock_code: string
  alert_type: 'price_above' | 'price_below'
  threshold: number
  enabled: boolean
  created_at: string
}

export interface AlertRecord {
  id: number
  rule_id: number | null
  stock_code: string
  triggered_at: string
  price: number | null
  message: string | null
}
